zospy/zpcore.py

```diff
@@ -282,13 +282,21 @@ class ZOS:
     ----------
         ZOSAPI (None | netModuleObject): The ZOSAPI interface or if loaded.
         ZOSAPI_NetHelper (None | netModuleObject): The ZOSAPI_NetHelper interface if loaded.
+
+    Parameters
+    ----------
+    preload: bool
+        A boolean indicating if nested namespaces should be preloaded.
+    zosapi_nethelper: str, optional
+        Optional filepath to the ZOSAPI_NetHelper dll, if None, the windows registry will be used to find
+        ZOSAPI_NetHelper dll. Defaults to None.
     """
 
     _OpticStudioSystem = OpticStudioSystem
 
     _instances = set()
 
-    def __new__(cls, *args, **kwargs):
+    def __new__(cls, preload: bool = False, zosapi_nethelper: str = None, *args, **kwargs):
```

Comment:
These arguments are only needed in `__init__`, not in `__new__` (and are therefore not used in the body of this method). `__new__` is only used to create a new, uninitialized `ZOS` object. These parameters are only necessary during initialization of that object.

Reply:
I think we cannot directly drop them, as a `TypeError` would be raised [when either of these arguments is specified upon initiating `ZOS`](https://stackoverflow.com/questions/59217884/new-method-giving-error-object-new-takes-exactly-one-argument-the-typ). For example, `zos = ZOS(preload=True)` would raise a `TypeError`. 

This is the result of `*args` and `**kwargs` being passed to `instance = super(ZOS, cls).__new__(cls, *args, **kwargs)`, which results in the arguments of `__init__()` being passed when specified. As the super of ZOS is always `object()`, this raises the `TypeError`.

We could handle this elegantly by for example [checking wheter the base class is `object`](https://github.com/python/typing/pull/517), but I think we have a controlled situation and can simply adjust `instance = super(ZOS, cls).__new__(cls, *args, **kwargs)` to `instance = super(ZOS, cls).__new__(cls)`

I would like to use that option, as it will be able to add additional arguments to `ZOS.__init__()` in the future without it affecting `ZOS.__new__()`.

Reply:
Ah, I see I was mistaken. Your preferred option sounds good!

Reply:
I have adjusted this accordingly